### PR TITLE
Add plugins to documentation (and a few other documentation changes too)

### PR DESCRIPTION
**Pull request by ichard26**
Updated version here ~(currently a bit broken, see to-do list below)~:
https://ichard26-testbandersnatchdocs.readthedocs.io/en/add_plugins_to_docs/index.html

To-do list:
- [x] Add entries so bandersnatch plugins are picked up by autodoc
- [x] Fix installation and import failures
- [x] Fix epub build
  - [x] Ignore `unknown mimetype for '.nojekyll'` warning ~(unpushed yet)~
  - [x] Fill in the copyright field in `docs/conf.py`  - waiting for what to put here from cooperlees :) 
- [x] Look into and fix Travis CI failure ...
- [x] Final checks and maybe a merge from master

### CODE REVIEW

**Comment by cooperlees** [comment]
I guess add in Copyright to the PSF via the PyPA? What have other PyPA projects done. Lets just do the same :)

File: docs/conf.py [hunk 1]
```diff
@@ -23,6 +23,10 @@ class DocStub:
 # If your documentation needs a minimal Sphinx version, state it here.
 needs_sphinx = "3.0"
 
+# epub build warns about .nojekyll of which Sphinx creates on build,
+# so just ignore the warning since we can't do anything a-boat it :D
+suppress_warnings = ["epub.unknown_project_files"]
```
Comment (ichard26) [inline docs/conf.py:28]: spot the Canadian :D

File: requirements_docs.txt [hunk 2]
```diff
@@ -10,3 +10,9 @@ xmlrpc2==0.3.1
 
 git+https://github.com/pypa/pypa-docs-theme.git#egg=pypa-docs-theme
 git+https://github.com/python/python-docs-theme.git#egg=python-docs-theme
+
+# This is needed since autodoc imports all bandersnatch packages and modules
+# so imports must not fail or its containing module will NOT be documented.
+# Also, the missing swift dependencies will cause the doc build to fail since
+# autodoc will raise a warning due to the import failure.
+-r requirements_swift.txt
```
Comment (ichard26) [inline requirements_docs.txt:18]: This should be *fine* as long people are OK with installing 40+ dependencies (I didn't count, but it's a LOT)

**Comment by cooperlees** [comment]
Looks good, autodoc benefits outweights installing more deps slowness :)

**Comment by cooperlees** [comment]
https://bandersnatch.readthedocs.io/en/latest/ - Live.

**Pull request merged**